### PR TITLE
chore: bunch of small refactors for improved devex

### PR DESCRIPTION
**Pull request by samlaf**
See commit names for summary of changes.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -159,25 +155,26 @@ func (svr *Server) Health(w http.ResponseWriter, _ *http.Request) error {
 func (svr *Server) HandleGet(w http.ResponseWriter, r *http.Request) error {
 	ct, err := ReadCommitmentMode(r)
```
Comment (epociask) [inline server/server.go:156]: knit and irrelevant to PR - ct implies `CommitmentType` but it should probably shorthand notation `cm`

File: server/server.go [hunk 2]
```diff
@@ -25,13 +25,9 @@ var (
 )
 
 const (
-	invalidDomain         = "invalid domain type"
```
Comment (epociask) [inline server/server.go:28]: TY

File: cmd/server/main.go [hunk 3]
```diff
@@ -16,7 +16,9 @@ import (
 	"github.com/ethereum-optimism/optimism/op-service/opio"
 )
 
-var Version = "v1.1.0"
+var Version = "unknown"
+var Commit = "unknown"
+var Date = "unknown"
```
Comment (epociask) [inline cmd/server/main.go:21]: knit - could encapsulate under a single var declaration

**Review by epociask — APPROVED** [review]
LGTM - just a slew of knits but nothing blocking 😄

File: cmd/server/main.go (hunk 3 above)
Comment (samlaf) [reply cmd/server/main.go]: [0d05a6d](https://github.com/Layr-Labs/eigenda-proxy/pull/124/commits/0d05a6d7acca06cbf8519035abfd96f77d15270a)

File: server/server.go (hunk 1 above)
Comment (samlaf) [reply server/server.go]: [624c815](https://github.com/Layr-Labs/eigenda-proxy/pull/124/commits/624c815c35812c321596d21c0e1fee71f4e40d0c)

File: server/server.go [hunk 4]
```diff
@@ -205,22 +203,23 @@ func (svr *Server) HandlePut(w http.ResponseWriter, r *http.Request) error {
 	if len(key) > 0 && key != "put" { // commitment key already provided (keccak256)
 		comm, err = commitments.StringToDecodedCommitment(key, ct)
 		if err != nil {
-			svr.log.Info("failed to decode commitment", "err", err, "key", key)
-			w.WriteHeader(http.StatusBadRequest)
+			err = fmt.Errorf("failed to decode commitment from key %v (commitment mode %v): %w", key, ct, err)
```
[the remark this answers is not in the document]
Comment (samlaf) [reply server/server.go]: wdym by "into a single"?

[the remark this answers is not in the document]
Comment (samlaf) [reply server/server.go]: merging this to unblock other work, but ping me to let me know what you meant and we can create another PR

**Pull request merged**